core/Segment.php

```diff
@@ -180,17 +180,30 @@ public function getSegmentExpression()
         return $this->segmentExpression;
     }
 
+    /**
+     * @throws Exception
+     */
     private function getAvailableSegments()
     {
+        // start cache
+        $cache = PiwikCache::getTransientCache();
+
+        //covert cache id
+        $cacheId = 'API.getSegmentsMetadata.'.SettingsPiwik::getPiwikInstanceId() . '.' . implode(",", $this->idSites);
```

Comment:
just fyi, no change needed. For regular caches we don't need the instance ID in there. That was for the `get_lock` mysql method only.

Reply:
@tsteur ah, this is already merged, should I update this in another PR

Reply:
@peterhashair no changed needed (mention this in the comment above). It's just for future when we work with caches again.